src/game/TileEngine/Explosion_Control.cc

```diff
@@ -1168,11 +1168,58 @@ static BOOLEAN ExpAffect(const INT16 sBombGridNo, const INT16 sGridNo, const UIN
 	 }
 	 else
 	 {
-			SOLDIERTYPE* const pSoldier = WhoIsThere2(sGridNo, bLevel);
-			if (pSoldier == NULL) return fRecompileMovementCosts;
-			// someone is here, and they're gonna get hurt
+		 SOLDIERTYPE* const pSoldier = WhoIsThere2(sGridNo, bLevel);
+		 if (pSoldier == NULL) return fRecompileMovementCosts;
+		 // someone is here, and they're gonna get hurt
+
+		 // We check if we can get hit by gas
+		 // First check for gasmask on head
+		 INT8 bPosOfMask;
+		 if (pSoldier->inv[HEAD1POS].usItem == GASMASK && pSoldier->inv[HEAD1POS].bStatus[0] >= GASMASK_MIN_STATUS)
+		 {
+			 bPosOfMask = HEAD1POS;
+		 }
+		 else if (pSoldier->inv[HEAD2POS].usItem == GASMASK && pSoldier->inv[HEAD2POS].bStatus[0] >= GASMASK_MIN_STATUS)
+		 {
+			 bPosOfMask = HEAD2POS;
+		 }
+		 else
+		 {
+			 bPosOfMask = NO_SLOT;
+		 }
 
-			fRecompileMovementCosts = DishOutGasDamage(pSoldier, pExplosive, sSubsequent, fRecompileMovementCosts, sWoundAmt, sBreathAmt, owner);
+		 // Look for gas
+		 EXPLOSIVETYPE const* pExplosive = 0;
+		 if (!AM_A_ROBOT(pSoldier))
+		 {
+			 if (gpWorldLevelData[pSoldier->sGridNo].ubExtFlags[pSoldier->bLevel] & MAPELEMENT_EXT_TEARGAS)
+			 {
+				 if (!(pSoldier->fHitByGasFlags & HIT_BY_TEARGAS) && bPosOfMask == NO_SLOT)
+				 {
+					 // check for gas mask
```

Comment:
Shouldn't this comment be higher up?
Also, saving `gpWorldLevelData[pSoldier->sGridNo].ubExtFlags[pSoldier->bLevel]` to a temporary var would make this whole section more readable.